src/context/context.rs

```diff
@@ -0,0 +1,558 @@
+    /// #     let formatter = Arc::new(Mutex::new(simple));
+    /// #     let configuration = Configuration::default().parallel(false);
+    /// #     let runner = Runner::new(configuration, vec![formatter]);
+    /// #
+    /// runner.run_or_exit(rspec::suite("a test suite", (), |ctx| {
+    ///     ctx.context("opens a context labeled 'context'", |ctx| {
+    ///         // …
+    ///     });
+    /// }));
+    /// # }
+    /// ```
+    ///
+    /// Corresponding console output:
+    ///
+    /// ```no-run
+    /// running tests
+    /// Suite "a test suite":
+    ///     Context "opens a sub-context":
+    ///         …
+    /// ```
+    ///
+    /// Available aliases:
+    ///
+    /// - [`specify`](struct.Context.html#method.specify).
+    /// - [`when`](struct.Context.html#method.when).
+    pub fn context<'a, S, F>(&mut self, name: S, body: F)
+    where
+        S: Into<Option<&'a str>>,
+        F: FnOnce(&mut Context<T>) -> (),
+        T: ::std::fmt::Debug,
+    {
+        let info = name.into().map(|name| {
+            ContextInfo {
+                label: ContextLabel::Context,
+                name: name.to_owned(),
+            }
+        });
+        self.context_internal(info, body)
+    }
+
+    /// Alias for [`context`](struct.Context.html#method.context), see for more info.
+    ///
+    /// Available further aliases:
+    ///
+    /// - [`when`](struct.Context.html#method.when).
+    pub fn specify<'a, S, F>(&mut self, name: S, body: F)
+    where
+        S: Into<Option<&'a str>>,
+        F: FnOnce(&mut Context<T>) -> (),
+        T: ::std::fmt::Debug,
+    {
+        let info = name.into().map(|name| {
+            ContextInfo {
+                label: ContextLabel::Specify,
+                name: name.to_owned(),
+            }
+        });
+        self.context_internal(info, body)
+    }
+
+    /// Alias for [`context`](struct.Context.html#method.context), see for more info.
+    ///
+    /// Available further aliases:
+    ///
+    /// - [`specify`](struct.Context.html#method.specify).
+    pub fn when<'b, S, F>(&mut self, name: S, body: F)
+    where
+        S: Into<Option<&'b str>>,
+        F: FnOnce(&mut Context<T>) -> (),
+        T: ::std::fmt::Debug,
+    {
+        let info = name.into().map(|name| {
+            ContextInfo {
+                label: ContextLabel::When,
+                name: name.to_owned(),
+            }
+        });
+        self.context_internal(info, body)
+    }
+
+    /// Open a new name-less context within the current context which won't show up in the logs.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate rspec;
+    /// #
+    /// # use std::io;
+    /// # use std::sync::{Arc, Mutex};
+    /// #
+    /// # use rspec::prelude::*;
+    /// #
+    /// # pub fn main() {
+    /// #     let simple = rspec::formatter::Simple::new(io::stdout());
+    /// #     let formatter = Arc::new(Mutex::new(simple));
```

Comment:
thought: if the formatter needs `Arc<Mutex<F>`, why not put it in the formatter itself ?